ddtrace/appsec/_patch_subprocess_executions.py

```diff
@@ -0,0 +1,400 @@
+import collections
+from fnmatch import fnmatch
+import os
+import re
+import shlex
+import subprocess
+from threading import RLock
+from typing import Deque
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+from typing import cast
+
+import attr
+import six
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.appsec._constants import COMMANDS
+from ddtrace.contrib import trace_utils
+from ddtrace.ext import SpanTypes
+from ddtrace.internal import _context
+from ddtrace.internal.compat import PY2
+from ddtrace.internal.compat import shjoin
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def _patch():
+    # type: () -> None
+    if not config._appsec_enabled:
+        return
+
+    import os
+
+    Pin().onto(os)
+    trace_utils.wrap(os, "system", traced_ossystem(os))
```

Comment:
We know that some frameworks do monkey-patching (e.g. gevent does monkey-patching of the os module). What are the odds that such frameworks would undo our instrumentation?

Reply:
Well, high I guess, specially if the patching is done after we have done ours. This is something that probably should be documented. For gevent, we can detect the patching and at least write a warning, but I'll investigate if there are better options (if you have ideas since you've fought a lot with gevent I'm all ears!).

Reply:
Another update:

Tested with a simple flask app, ran with gevent. Seems to work but of course in this case gevent is run before ddtrace, which matches the case that worked in my first test.

Reply:
> if you have ideas since you've fought a lot with gevent I'm all ears!

I think this is tricky to deal with. Normally we would want to trigger patching only after a module that needs patching is imported. But if we do this here we would still end up with gevent undoing our monkey-patching (we would patch soon after gevent imports the module that it wants to patch, and then gevent would replace things, thus removing our patching).